src/lexer/token.h

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include <string>
+
+namespace lexer {
+enum class TokenType {
+  END_OF_FILE,
+  // Identifiers.
+  LOWER_CASE_IDENT,
+  MACRO_IDENT,
+  UPPER_CASE_IDENT,
+  // Literals.
+  INT,
+  HEX,
+  OCT,
+  BINARY_NUMBER,
+  STRING,
+  RAWSTRING,
+  CODESTRING,
+  RAWCODESTRING,
+  // Binary operators.
+  PLUS,
+  MINUS,
+  DIVIDE,
+  STAR,
+  OR,
+  AND,
+  GREATER,
+  LESS,
+  EQUAL,
+  DIFFERENT,
+  BITSHIFT_RIGHT,
+  BITSHIFT_LEFT,
+  BITOR,
+  BITAND,
+  BITXOR,
+  ASSIGN,
+  PLUS_ASSIGN,
+  MINUS_ASSIGN,
+  TIMES_ASSIGN,
+  DIVIDE_ASSIGN,
+  XOR_ASSIGN,
+  OR_ASSIGN,
+  AND_ASSIGN,
+  // Unary operators.
+  AMPERSAND,
+  BANG,
+  TILDE,
+  // Delimiters.
+  OPEN_PAREN,
+  CLOSE_PAREN,
+  OPEN_BRACKET,
+  CLOSE_BRACKET,
+  OPEN_BRACE,
+  CLOSE_BRACE,
+  SEMICOLON,
+  COLON,
+  COMMA,
+  ARROW,  // ->
+  // Keywords (alphabetical order).
+  AS,
+  DO,
+  ENUM,
+  DATA,
+  CLASS,
+  PRIVATE,
+  PUBLIC,
+  FROM,
+  FOR,
+  FORWARD,  // for forward-references.
+  FUN,
+  IF,
+  IMPORT,
+  IS,
+  MUT,
+  RETURN,
+  VAL,
+  WHEN,
+  WHILE,
+  // Mics.
+  ANNOTATION,
+  MACRO_CALL,
+};
+
+struct Location {
```

Comment:
I mean, you have fields line and column in the Lexer class and then you are using them to populate a Location in the Token result. You could use directly a Location in the Lexer class.